### PR TITLE
Allow FP16 support in tensor core.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

**Comment by dlibenzi** [comment]
Fixes #1936

File: torch_xla/csrc/tensor_util.cpp [hunk 1]
```diff
@@ -829,11 +882,15 @@ xla::PrimitiveType GetDevicePrimitiveType(xla::PrimitiveType type,
...
+      if (UseF16()) {
+        return xla::PrimitiveType::F16;
+      }
       return xla_device.hw_type != DeviceType::TPU ? xla::PrimitiveType::F64
                                                    : xla::PrimitiveType::F32;
     case xla::PrimitiveType::F32:
-      // When PyTorch will support native BF16 type, the global configuration
-      // can be replaced (or augmented) with the proper mapping.
+      if (UseF16()) {
+        return xla::PrimitiveType::F16;
```
Comment (JackCaoG) [inline torch_xla/csrc/tensor_util.cpp:892]: Do we have a way to block user from setting both USE_F16 and USE_BF16 to 1? If user (mistakenly) do that, for F64 we will use BF16 and for F32 we will use F16. Might be better if we are consistently use one of them by switching the order of if check?

Comment (dlibenzi) [reply torch_xla/csrc/tensor_util.cpp:892]: That will result in errors in any case, as no platforms supports both 😉 
But for code consistency sake makes sense to flip the test.

**Comment by JackCaoG** [comment]
I will find sometime to go over the python test we disabled due to `half not supported`, should be able to enable some of them

**Comment by dlibenzi** [comment]
> I will find sometime to go over the python test we disabled due to `half not supported`, should be able to enable some of them

FP16 is like BF16, we will unlikely find any support on XLA CPU.
FP16 is only for XLA GPU. Like BF16 is only for XLA TPU.

**Comment by JackCaoG** [comment]
> > I will find sometime to go over the python test we disabled due to `half not supported`, should be able to enable some of them
> 
> FP16 is like BF16, we will unlikely find any support on XLA CPU.
> FP16 is only for XLA GPU. Like BF16 is only for XLA TPU.

Ah I see, thanks for the info

**Pull request merged**